Tools/Scripts/libraries/webkitcorepy/webkitcorepy/docker.py

```diff
@@ -0,0 +1,284 @@
+# Copyright (C) 2023 Apple Inc. All rights reserved.
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions
+# are met:
+# 1.  Redistributions of source code must retain the above copyright
+#    notice, this list of conditions and the following disclaimer.
+# 2.  Redistributions in binary form must reproduce the above copyright
+#    notice, this list of conditions and the following disclaimer in the
+#    documentation and/or other materials provided with the distribution.
+#
+# THIS SOFTWARE IS PROVIDED BY APPLE INC. AND ITS CONTRIBUTORS ``AS IS'' AND ANY
+# EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+# WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+# DISCLAIMED. IN NO EVENT SHALL APPLE INC. OR ITS CONTRIBUTORS BE LIABLE FOR ANY
+# DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+# (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+# LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON
+# ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+# (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+# SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+import os
+import sys
+import time
+
+from webkitcorepy import decorators, run
+
+
+class Docker(object):
+    _installed = None
+    _command = None
+    DOCKER_APP = '/Applications/Docker.app'
+    TIMEOUT = 10
+
+    class Container(object):
+        def __init__(self, image, name=None, tag='latest', remove=True, ports=None, port=None, volumes=None):
+            self.image = image
+            self.name = name or image
+            self.tag = tag
+
+            self.remove = remove
+            self.owner = False
+            self.ports = ports or {}
+            self.volumes = volumes or []
+            if port:
+                self.ports[port] = port
+
+        def __enter__(self, cached=True):
```

Comment:
If `__enter__` is meant for context manager, I'm not aware of a way to run with different `cached` value other than true. 
Also, it looks like almost all functions in this class and `Volume` takes `cached` as argument, I wonder if there is a better way.